inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_matmul_to_fc_or_gemm.cpp

```diff
@@ -150,54 +150,82 @@ ngraph::pass::ConvertMatMulToFCorGemm::ConvertMatMulToFCorGemm() {
 
             ngraph::copy_runtime_info(matmul, new_ops);
             ngraph::replace_node(matmul, fc);
-        } else {
-            // WA for IE that Gemm must have inputs with the same length.
-            if (shape_a.size() < shape_b.size()) {
-                // Reshape first input (fc_input_a)
-                Shape reshape_shape(shape_b.size() - shape_a.size(), 1);
-                reshape_shape.insert(reshape_shape.end(), shape_a.begin(), shape_a.end());
-                fc_input_a = op::util::reshapeTo(fc_input_a, reshape_shape);
-                new_ops.push_back(fc_input_a.get_node_shared_ptr());
-            } else if (shape_b.size() < shape_a.size()) {
-                // Reshape second input (fc_input_b)
-                Shape reshape_shape;
-                if (shape_b.size() == 1) {
-                    // In case if shape_b has only one dimension we reshape it to [...,1,X,1]
-                    reshape_shape = Shape(shape_a.size() - (shape_b.size() + 1), 1);
-                    reshape_shape.push_back(shape_b[0]);  // add X dimension
-                    reshape_shape.push_back(1);  // add last 1 dimension
-                } else {
-                    // In this case we reshape shape_b to [...,1,1,X]
-                    reshape_shape = Shape(shape_a.size() - shape_b.size(), 1);
-                    reshape_shape.insert(reshape_shape.end(), shape_b.begin(), shape_b.end());
-                }
-                fc_input_b = op::util::reshapeTo(fc_input_b, reshape_shape);
-                new_ops.push_back(fc_input_b.get_node_shared_ptr());
-            }
+            return true;
+        }
+        return false;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(matmul, "ConvertMatMulToFC");
+    this->register_matcher(m, callback);
+}
+
+ngraph::pass::ConvertMatMulToGemm::ConvertMatMulToGemm() {
+    auto input_0 = std::make_shared<pattern::op::Label>(element::f32, Shape {1, 1});
+    auto input_1 = std::make_shared<pattern::op::Label>(element::f32, Shape {1, 1});
+    auto matmul = std::make_shared<ngraph::opset1::MatMul>(input_0, input_1);
+
+    ngraph::matcher_pass_callback callback = [this](pattern::Matcher& m) {
+        auto matmul = std::dynamic_pointer_cast<ngraph::opset1::MatMul>(m.get_match_root());
+        if (!matmul) {
```

Comment:
Why you don't check predicate for `ConvertMatMulToGemm`?

Reply:
Because in this case it will disable conversion to gemm also. It's current limitations of predicate and I'm working on its expansion.